api/src/utils/media/csv/validation/csv-row-validator.ts

```diff
@@ -62,127 +59,159 @@ export const getRequiredFieldsValidator = (requiredFieldsByHeader?: string[]): C
  *
- * @param {ICodeValuesByHeader[]} [codeValuesByHeader]
+ * @param {ColumnRangeValidatorConfig} [config]
  * @return {*}  {CSVValidator}
  */
-export const getValidRangeFieldsValidator = (requiredRangeByHeader?: IValueRangesByHeader[]): CSVValidator => {
+export const getValidRangeFieldsValidator = (config?: ColumnRangeValidatorConfig): CSVValidator => {
   return (csvWorksheet) => {
-    if (!requiredRangeByHeader) {
+    if (!config) {
       return csvWorksheet;
     }
 
     const rows = csvWorksheet.getRows();
     const headers = csvWorksheet.getHeaders();
 
     rows.forEach((row, rowIndex) => {
-      for (const valueRangesByHeader of requiredRangeByHeader) {
-        const columnIndex = headers.indexOf(valueRangesByHeader.header);
-
-        //if column does not exist, return
+      const columnIndex = headers.indexOf(config.columnName);
 
-        if (columnIndex < 0) {
-          return csvWorksheet;
-        }
+      // if column does not exist, return
+      if (columnIndex < 0) {
+        return csvWorksheet;
+      }
 
-        const rowValueForColumn = Number(row[columnIndex]);
+      const rowValueForColumn = Number(row[columnIndex]);
+
+      if (isNaN(rowValueForColumn)) {
+        csvWorksheet.csvValidation.addRowErrors([
+          {
+            errorCode: 'Invalid Value',
+            message: `Invalid value: ${row[columnIndex]}. Value must be a number `,
+            col: config.columnName,
+            row: rowIndex + 2
+          }
+        ]);
+      }
 
-        if (isNaN(rowValueForColumn)) {
+      if (config.column_range_validator.min_value && config.column_range_validator.max_value) {
+        // Value must be between min value and max value
+        if (
+          rowValueForColumn < config.column_range_validator.min_value ||
+          rowValueForColumn > config.column_range_validator.max_value
+        ) {
+          // Add an error if the cell value is not in the correct range provided in the array
           csvWorksheet.csvValidation.addRowErrors([
             {
-              errorCode: 'Invalid Value',
-              message: `Invalid value: ${row[columnIndex]}. Value must be a number `,
-              col: valueRangesByHeader.header,
+              errorCode: 'Out of Range',
+              message: `Invalid value: ${rowValueForColumn}. Value must be between ${config.column_range_validator.min_value} and ${config.column_range_validator.max_value} `,
+              col: config.columnName,
               row: rowIndex + 2
             }
           ]);
         }
-
-        // Add an error if the cell value is not in the correct range provided in the array
-
-        if (rowValueForColumn < valueRangesByHeader.min_value || rowValueForColumn > valueRangesByHeader.max_value) {
+      } else if (!config.column_range_validator.min_value && config.column_range_validator.max_value) {
+        // Value must be less than max value
+        if (rowValueForColumn > config.column_range_validator.max_value) {
```

Comment:
question: the comment says must be less than but the check is for greater than and not for greater than or equal to
maybe either the check or the comment needs to be updated?

Reply:
The comment is maybe just a bit confusing.

The value does indeed need to be less than the max value.  If it is greater, inside that if we add an error.